src/App/PropertyOverrides.h

```diff
@@ -0,0 +1,292 @@
+     *
+     * @param prop Reference to the property being overridden.
+     * @param newValue The temporary value to apply.
+     */
+    ScopedPropertyOverride(P& prop, const Value& newValue)
+        : _prop(&prop)
+        , _old(PropAccessor<P>::get(prop))
+        , _active(true)
+    {
+        PropAccessor<P>::set(*_prop, newValue);
+    }
+
+    FC_DISABLE_COPY(ScopedPropertyOverride);
+
+    ScopedPropertyOverride(ScopedPropertyOverride&& other) noexcept
+        : _prop(other._prop)
+        , _old(std::move(other._old))
+        , _active(other._active)
+    {
+        other._prop = nullptr;
+        other._active = false;
+    }
+
+    ScopedPropertyOverride& operator=(ScopedPropertyOverride&& other) noexcept
+    {
+        if (this != &other) {
+            restore();
+            _prop = other._prop;
+            _old = std::move(other._old);
+            _active = other._active;
+            other._prop = nullptr;
+            other._active = false;
+        }
+        return *this;
+    }
+
+    /**
+     * @brief Destructor.
+     *
+     * Restores the original value of the property.
+     */
+    ~ScopedPropertyOverride()
+    {
+        restore();
+    }
+
+    /**
+     * @brief Checks if the guard is currently active.
+     *
+     * @return True if the override is in effect.
+     */
+    bool engaged() const
+    {
+        return _active;
+    }
+
+private:
+    void restore() noexcept
+    {
+        if (_prop && _active) {
+            PropAccessor<P>::set(*_prop, _old);
+            _active = false;
+        }
+    }
+
+    P* _prop {nullptr};    ///< Pointer to the property being overridden.
+    Value _old {};         ///< Saved old value.
+    bool _active {false};  ///< Whether this guard is currently active.
+};
+
+/**
+ * @brief Helper function to create a `ScopedPropertyOverride` with type deduction.
+ *
+ * @tparam P The property type.
+ * @tparam V The value type (deduced).
+ * @param prop Property reference.
+ * @param v New value to assign temporarily.
+ *
+ * @return A `ScopedPropertyOverride` guard.
+ */
+template<PropertyLike P, class V>
+auto makeOverride(P& prop, V&& v)
+{
+    using Guard = ScopedPropertyOverride<P>;
+    return Guard(prop, static_cast<Guard::Value>(std::forward<V>(v)));
+}
+
+/**
+ * @brief RAII context for managing multiple property overrides.
+ *
+ * `PropertyOverrideContext` stores multiple property overrides at once. When the
+ * context is destroyed, all properties are restored in reverse (LIFO) order.
+ *
+ * ### Example
+ * @code{.cpp}
+ * App::PropertyBool& vis = ...;
+ * App::PropertyString& label = ...;
+ *
+ * App::PropertyOverrideContext overrides;
+ * overrides.override(vis, false);
+ * overrides.override(label, "Temp");
+ * // both props overridden
+ * // ...
+ * // restored automatically when ctx is destroyed
+ * @endcode
+ */
+class PropertyOverrideContext
+{
+public:
+    PropertyOverrideContext() = default;
+
+    FC_DISABLE_COPY(PropertyOverrideContext);
```

Comment:
If you're going to use the macro, then in this case I'd go with the combo `FC_DISABLE_COPY_MOVE`.

Reply:
I'd do that if they both were disable but that class is non-copyable and moveable.